setup.py

```diff
@@ -1,9 +1,46 @@
 #!/usr/bin/env python
 
 import os
+import sys
 from setuptools import setup
 import versioneer
 
+
+CURRENT_PYTHON = sys.version_info[:2]
+REQUIRED_PYTHON = (3, 5)
+
+# Borrowed from
+# https://github.com/django/django/commit/32ade4d73b50aed77efdb9dd7371c17f89061afc
+# This check and everything above must remain compatible with Python 2.7.
+if CURRENT_PYTHON < REQUIRED_PYTHON:
+    sys.stderr.write(
+        """
+==========================
+Unsupported Python version
+==========================
+
+This version of distributed requires Python {}.{}, but you're trying to
+install it on Python {}.{}.
+
+This may be because you are using a version of pip that doesn't
+understand the python_requires classifier. Make sure you
+have pip >= 9.0 and setuptools >= 24.2, then try again:
+
+    $ python -m pip install --upgrade pip setuptools
+    $ python -m pip install distributed
+
+This will install the latest version of distributed that works on your
+version of Python. If you can't upgrade your pip (or Python), request
+an older version of distributed:
+
+    $ python -m pip install "distributed<2"
+""".format(
+            *(REQUIRED_PYTHON + CURRENT_PYTHON)
+        )
+    )
+    sys.exit(1)
```

Comment:
Do we actually need to do this though?  I would have thought that the `python_requires=` keyword would suffice.

Reply:
The behavior with a modern version of pip on Python 2 for `dask`seems sensible to me without this.